---
layout: default
title: Overview
nav_order: 1
description: "Overview of Restropective saddlepoint approximation (SPA) methods in GWAS."
has_children: false
has_toc: false
---

# Overview of restropective saddlepoint approximation (SPA) methods in GWAS

The restropective saddlepoint approximation (SPA) methods are mainly designed to conduct **genome-wide association studies (GWAS)** in terms of both single-variant and set-based analysis. 

## Genome-wide association studies

All retrospective-SPA-based GWAS approaches share the same analysis framework including the following two steps

- Step 1: Fit a null model using trait, covariates, and GRM (if applied).
- Step 2: Conduct single-variant or set-based tests to identify marker or marker-set (e.g. gene) significantly associated with the trait of interest.


##  Genome-wide gene-environment interaction (GxE) studies

All retrospective-SPA-based GxE approaches share the same analysis framework including the following two steps

- Step 1: Fit a genotype-independent (covariates-only) model using trait, covariates, and GRM (if applied).
- Step 2: Conduct single-variant or set-based tests to identify marker or marker-set (e.g. gene) significantly associated with the trait of interest.



# Restropective SPA Project Timeline

```markdown
# Timeline of My Academic and Technical Adventures

## March 2021
- The idea of a retrospective saddlepoint approximation (SPA) was conceived.

## October 2021
- The SPAmix algorithm was successfully designed.

## October 2021 - February 2022
- Work on the SPAGxE algorithm was initiated.

## May 2022
- Master's thesis 'Research on Empirical Saddlepoint Approximation Methods and Their Applications in Genome-Wide Association Studies' was completed.
```








